### PR TITLE
Add test about amtool minimal version

### PR DESCRIPTION
**Pull request by rporres**
Or tests silently fail not giving information about the issue.

Signed-off-by: Rafa Porres Molina <rporresm@redhat.com>

### CODE REVIEW

**Comment by chassing** [comment]
Please also enable full type hints check in `setup.cfg`
```
[mypy-reconcile.test.test_amtool]
check_untyped_defs = True
disallow_incomplete_defs = True
```

File: reconcile/utils/amtool.py [hunk 1]
```diff
@@ -45,6 +46,19 @@ def config_routes_test(yaml_config: str, labels: Mapping[str, str]) -> AmtoolRes
...
 
 
+def version() -> AmtoolResult:
+    """Returns the version parsed from amtool --version"""
+    result = _run_cmd(["amtool", "--version"])
+
+    pattern = re.compile("^amtool, version (?P<version>[^ ]+) .+")
+    if m := pattern.match(result.message):
+        return AmtoolResult(True, m.group("version"))
+    else:
```
Comment (chassing) [inline reconcile/utils/amtool.py:56]: this `else` branch isn't needed

**Review by chassing — APPROVED** [review]
Some nits. Otherwise LGTM

File: reconcile/utils/amtool.py (hunk 1 above)
Comment (rporres) [reply reconcile/utils/amtool.py]: ah, yes, return false by default. Sure that's more idiomatic, fixed in d31e81f4

**Pull request merged**